### PR TITLE
Feature/quagga

### PR DESCRIPTION
**Pull request by azasdimi**
Created the barcode scanner using QuaggaJS

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
     "gatsby-source-filesystem": "2.1.22",
     "gatsby-transformer-sharp": "2.2.14",
     "prop-types": "15.7.2",
+    "quagga": "^0.12.1",
```
Comment (RogierIBM) [inline frontend/package.json]: Please remove ^

File: frontend/src/components/scan-product/scanProduct.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        {this.state.scanning ? (
+          <Scanner onDetected={this.state._onDetected} />
+        ) : null}
+      </div>
+    )
+  }
+}
+export default ScanProductComponent
+
+// <ul className="results">
```
Comment (RogierIBM) [inline frontend/src/components/scan-product/scanProduct.js]: Is this needed?

File: frontend/src/components/scan-product/scanner.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      locate: true,
+    }, function (err) {
+      if (err) {
+        console.log(err);
+        return
+      }
+      console.log("Initialization finished. Ready to start");
+      Quagga.start();
+    });
+    console.log("component did mount.")
```
Comment (RogierIBM) [inline frontend/src/components/scan-product/scanner.js]: Console log needed?

File: frontend/src/components/scan-product/scanner.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      decoder: {
+        readers: ['ean_reader'],
+      },
+      locate: true,
+    }, function (err) {
+      if (err) {
+        console.log(err);
+        return
+      }
+      console.log("Initialization finished. Ready to start");
```
Comment (RogierIBM) [inline frontend/src/components/scan-product/scanner.js]: Console log needed?

File: frontend/src/components/scan-product/scanner.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      Quagga.start();
+    });
+    console.log("component did mount.")
+    Quagga.onDetected(this._onDetected)
+  }
+  componentWillUnmount() {
+    Quagga.offDetected(this._onDetected)
+  }
+  _onDetected(result) {
+    console.log("ok1")
```
Comment (RogierIBM) [inline frontend/src/components/scan-product/scanner.js]: Please remove

File: frontend/src/components/scan-product/scanner.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    console.log(result.codeResult.code);
```
Comment (RogierIBM) [inline frontend/src/components/scan-product/scanner.js]: Please remove

File: frontend/src/components/scan-product/scanner.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    if (result) {
+      Quagga.offDetected(this._onDetected);
+    }
+  }
+
+
```
Comment (RogierIBM) [inline frontend/src/components/scan-product/scanner.js]: Remove white lines only one is needed

File: frontend/package.json (hunk 1 above)
Comment (azasdimi) [reply frontend/package.json]: Done

**Pull request merged**